subsys/caf/modules/ble_adv.c

```diff
@@ -98,9 +98,9 @@ static int settings_set(const char *key, size_t len_rd, settings_read_cb read_cb
 	return 0;
 }
 
-#ifdef COFNIG_CAF_BLE_ADV_DIRECT_ADV
-SETTINGS_STATIC_HANDLER_DEFINE(ble_adv, MODULE_NAME, NULL, settings_set, NULL, NULL);
-#endif /* CONFIG_CAF_BLE_ADV_DIRECT_ADV */
+if (IS_ENABLED(CONFIG_CAF_BLE_ADV_DIRECT_ADV)) {
```

Comment:
`if (IS_ENABLED(x))` is to be used inside function body. You can use normal preprocessor code here.

The idea behind `IS_ENABLED` is to always compile excluded code so it is checked for errors. When you use preprocessor the code is excluded before compilation is done. There is so no checks done against the code correctness. `IS_ENABLED` does not really exclude the code from compilation but instead depends on optimizer to remove a dead code when object file is constructed.
This said there is no reason to use `IS_ENABLED` instead of preprocessor outside of function body. The only reason would be the consistency. Still you should use it as `#if IS_ENABLED(x)`. I don't expect anything reasonable will come from placing the code as it was done here. It would be like placing a normal if statement outside of a function.

In summary please use:
```
#if CONFIG_CAF_BLE_ADV_DIRECT_ADV
...
#endif
```
```